doc/tutorial/alignment_file/index.md

```diff
@@ -0,0 +1,309 @@
+
+## Reading custom fields
+
+In many cases you are not interested in all of the information in a file.
+For this purpose, we provide the possibility to select specific seqan3::field's for a file.
+The file will read only those fields and fill the record accordingly.
+
+You can select fields by providing a seqan3::fields object as an extra parameter to the constructor:
+
+\snippet doc/tutorial/sequence_file/alignment_file_snippets.cpp read_custom_fields
+
+\attention The order in which you specify the selected fields determines the order of elements in the seqan3::record.
+
+In the example above we only select the id, sequence and flag information
+so the seqan3::record object has three tuple elements that are decomposed using structural bindings.
+
+Note that this is possible for all seqan3 file objects.
+
+\assignment{Exercise: Accumulating mapping qualities}
+
+Let's assume we want to compute the average mapping quality of a SAM file.
+
+For this purpose, write a small program that
+    * only reads the mapping quality (field::MAPQ) out of a SAM file and
+    * computes the average of all qualities.
+
+Use the following file to test your program:
+
+\snippet doc/tutorial/alignment_file/alignment_file_solution1.cpp sam_file
+
+It should output:
+```
+Average: 27.4
+```
+
+\endassignment
+\solution
+
+\snippet doc/tutorial/alignment_file/alignment_file_solution1.cpp solution
+
+\endsolution
+
+# Alignment representation in the SAM format
+
+In SeqAn3 we represent an alignment as a tuple of two *aligned_sequences*,
+as you have probably learned by now from the alignment tutorial.
+
+The SAM format is the common output format of read mappers where you align short read sequence
+to one or more large reference sequences.
+In fact, the SAM format stores those alignment information only partially:
+It **does not store the reference sequence** but only the read sequence and a *CIGAR* string
+representing the alignment based on the read.
+
+Take this SAM record as an example:
+
+```
+r003   73   ref   3   17   1M1D4M   *  0   0  TAGGC   *
+```
+
+The record gives you the following information:
+A read with name `r003` has been mapped to a reference with name `ref` at position `3`
+(in the reference, counting from 1) with a quality of `17` (phred scaled).
+The flag has a value of `73` which indicates that the read is paired, the first in pair, but the mate is unmapped
+(see [this website](https://broadinstitute.github.io/picard/explain-flags.html) for a nice explanation of SAM flags).
+Fields set to `0` or `*` are defaulted and contain no information.
+
+The cigar string is `1M1D4M` which represents the following alignment:
+
+```
+      1 2 3 4 5 6 7 8 9 ...
+ref   N N N N N N N N N ...
+read      T - A G G C
+```
+
+where the reference sequence is not known (represented by `N`).
+You will learn in the next section how to handle additional reference sequence information.
```

Comment:
Maybe give a link to some external resource on cigar strings?

Reply:
I linked the  [SAM specifications](https://samtools.github.io/hts-specs/SAMv1.pdf) and the [SAMtools paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2723002/).
Since all other articles I found are either a copy of the specs or questions and answers in forums.